### ISSUE TITLE
Why does XLA enable autocast so much slower than CUDA

### ISSUE THREAD

**Issue by ZhongYFeng** [pytorch/xla#3103]
## ❓ Questions and Help
I run both xla and cuda training rates with mixed precision enabled using the ResNet50 model, and found that xla was significantly slower than cuda. I don't know why this is happening, and the kernel time XLA is also slower than cuda
## cuda amp train
**code:**
net = ResNet50()
device = 'cuda' if torch.cuda.is_available() else 'cpu'
def train():
    net.train()
    for batch_idx, (inputs, targets) in enumerate(trainloader):
        start_time = time.time()
        inputs, targets = inputs.to(device), targets.to(device)
        with autocast():
            outputs = net(inputs)
            loss = criterion(outputs, targets)
        optimizer.zero_grad()
        scaler.scale(loss).backward()
        scaler.step(optimizer)
        scaler.update()
        if batch_idx == 10:
            break
        end_time = time.time()
        print("step %d: step_time: %.5f" % (batch_idx, end_time-start_time))
**result:**
step 0: step_time: 7.58732
step 1: step_time: 0.05985
step 2: step_time: 0.06600
step 3: step_time: 0.06300
step 4: step_time: 0.06158
step 5: step_time: 0.06120
step 6: step_time: 0.05985
step 7: step_time: 0.05543
step 8: step_time: 0.05959
step 9: step_time: 0.05963

##  xla amp train
**code:**
net = ResNet50()
device = xm.xla_device(devkind='GPU')
def train():
    net.train()
    for batch_idx, (inputs, targets) in enumerate(trainloader):
        start_time = time.time()
        inputs, targets = inputs.to(device), targets.to(device)
        with autocast():
            outputs = net(inputs)
            loss = criterion(outputs, targets)
        optimizer.zero_grad()
        scaler.scale(loss).backward()
        scaler.step(optimizer)
        scaler.update()
        xm.mark_step()
        if batch_idx == 10:
            break
        end_time = time.time()
        print("step %d: step_time: %.5f" % (batch_idx, end_time-start_time))
**result:**
step 0: step_time: 49.05568
step 1: step_time: 47.84207
step 2: step_time: 24.68861
step 3: step_time: 27.74563
step 4: step_time: 0.24843
step 5: step_time: 0.23068
step 6: step_time: 0.16089
step 7: step_time: 0.14444
step 8: step_time: 0.24297
step 9: step_time: 0.23277

## My doubts
According to the results XLA is about twice as slow as CUDA
Finally, I also had the nvprof tool analyze the kernel call times for each step, and cuda_amp took 55.117ms and XLA_amp 74.680ms in total.
I want to know why it is so slow. I have tested before that the difference between the two is very small without enable autocast.
Can someone analyze it for me? Thank you

**Comment by ZhongYFeng** [comment]
When judging whether the weight can be updated, the _maybe_opt_step method is called to determine whether there is INF or NAN. Here is a to_cpu operation, the graph will be compiled and executed once, and at the same time the operator aten::_local_scalar_dense will also be called. There is no lower for this operator. Therefore, every  step, it will consume a lot of time. Is there any better way to replace the implementation here, because it is too slow to update like this.
`    
def _maybe_opt_step(self, optimizer, optimizer_state, *args, **kwargs):
        retval = None
        if not sum(v.item() for v in optimizer_state["found_inf_per_device"].values()):
            retval = optimizer.step(*args, **kwargs)
        return retval
`

**Comment by JackCaoG** [comment]
My guess is that you are hitting those pt/xla common issues. For more info, you can read [this doc](https://github.com/pytorch/xla/blob/master/TROUBLESHOOTING.md). If all you want is resnet50 with amp, you can use our [code](https://github.com/pytorch/xla/blob/master/test/test_train_mp_imagenet.py) with `python test_train_mp_imagenet.py --fake_data --batch_size 128 --amp`. We have observed ~50% speed up on resnet with PyTorch/XLA:GPU.
